Add reverse_order option

frvade opened the pull request:
Added an option for reverse ordering.
Wrapped specs in transaction for keeping records intact after changes.

frvade committed:
add reverse_order option
fixed transaction absence in specs

frvade committed:
add documentation

frvade committed:
reverse_order → order

frvade committed:
fix readme

AnotherRegularDude commented on lib/sequel/extensions/batches/yielder.rb:
```diff
@@ -2,15 +2,20 @@
       self.start = start
       self.finish = finish
+
+      unless [:asc, :desc].include?(order)
```
`%i[asc desc]` - кажется, тру руби вей вот таким будет)

По идее, рубокоп должен ругаться на запись с несколькоми символами без `%i`

frvade replied:
поправил, рубокоп чето не заругался

frvade committed:
add enum when no block provided
github actions
remove travis
rubocop fixes
rubocop rake task extend
add bundler audit gem
remove coveralls gem

frvade committed:
fixed array of symbols

frvade committed:
add newline to ci.yml

tycooon commented on .rubocop.yml:
```diff
@@ -3,7 +3,16 @@ inherit_gem:
 AllCops:
   DisplayCopNames: true
-  TargetRubyVersion: 2.3
+  TargetRubyVersion: 2.5
```
2.5 уже официально не поддерживается

frvade replied:
поднял до 2.6

tycooon commented on lib/sequel/extensions/batches/yielder.rb:
```diff
@@ -2,33 +2,39 @@
+      self.order = options.delete(:order) || :asc
+      unless %i[asc desc].include?(order)
+        raise ArgumentError, ":order must be :asc or :desc, got #{order}"
+      end
```
Тут теперь неизвестные опции просто игнорятся, надо бы проверку впилить.

frvade replied:
впилил

frvade committed:
unknown options error
target ruby version to 2.6

tycooon commented:
Есть идеи почему воркфлоу не запустилось?

tycooon commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,48 @@
+name: CI
+
+on: [pull_request]
```
На пуш тоже неплохо бы гонять 🤔

frvade replied:
добавил

frvade committed:
push event workflow

frvade commented:
я подозреваю, что для запуска нужно, чтобы воркфлоу в мастере был. но это интуитивно, пруфов у меня нет

frvade commented:
а, вроде вот почему
https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows#pull-request-events-for-forked-repositories

frvade committed:
push event workflow

AnotherRegularDude commented:
> а, вроде вот почему https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows#pull-request-events-for-forked-repositories

Я как-то бомбил, что это лучшее, что смог придумать гитхаб для защиты от майнеров.
